lib/msal-react/src/hooks/useMsalAuthentication.ts

```diff
@@ -21,46 +21,56 @@ export function useMsalAuthentication(
     authenticationRequest?: PopupRequest|RedirectRequest|SsoSilentRequest, 
     accountIdentifier?: AccountIdentifiers
 ): MsalAuthenticationResult {
-    const isMounted = useRef(true);
-    useEffect(() => {
-        return () => {
-            isMounted.current = false;
-        };
-    }, []);
-
     const { instance, inProgress } = useMsal();
     const isAuthenticated = useIsAuthenticated(accountIdentifier);
     const [[result, error], setResponse] = useState<[AuthenticationResult|null, AuthError|null]>([null, null]);
     const [hasBeenCalled, setHasBeenCalled] = useState<boolean>(false);
 
-    const login = useCallback(async (loginType: InteractionType, request?: PopupRequest|RedirectRequest|SsoSilentRequest): Promise<AuthenticationResult|null> => {
+    const login = useCallback(async (callbackInteractionType?: InteractionType, callbackRequest?: PopupRequest|RedirectRequest|SsoSilentRequest): Promise<AuthenticationResult|null> => {
+        const loginType = callbackInteractionType || interactionType;
+        const loginRequest = callbackRequest || authenticationRequest || {};
         switch (loginType) {
             case InteractionType.Popup:
-                return instance.loginPopup(request as PopupRequest);
+                return instance.loginPopup(loginRequest as PopupRequest);
             case InteractionType.Redirect:
                 // This promise is not expected to resolve due to full frame redirect
-                return instance.loginRedirect(request as RedirectRequest).then(null);
+                return instance.loginRedirect(loginRequest as RedirectRequest).then(null);
             case InteractionType.Silent:
-                return instance.ssoSilent(request as SsoSilentRequest);
+                return instance.ssoSilent(loginRequest as SsoSilentRequest);
             default:
                 throw "Invalid interaction type provided.";
         }
+    }, [instance, interactionType, authenticationRequest]);
+
+    useEffect(() => {
+        const callbackId = instance.addEventCallback((message: EventMessage) => {
+            switch(message.eventType) {
+                case EventType.LOGIN_SUCCESS:
+                case EventType.SSO_SILENT_SUCCESS:
+                    message.payload && setResponse([message.payload as AuthenticationResult, null]);
```

Comment:
Nitpick (readability):
```suggestion
                    if (message.payload) {
                        setResponse([message.payload as AuthenticationResult, null]);
                    }
```

Reply:
Done